src/init.ts

```diff
@@ -4,146 +4,235 @@ import fs from 'fs';
-        type: 'input',
-        name: 'awsAccessKeyId',
-        message: 'Enter you AWS/Tigris Access Key Id:',
+        type: 'list',
+        name: 'selectedTransport',
+        message: 'Select Transport:',
+        choices: supportedTransports,
       },
     ]);
-    const { awsSecretAccessKey } = await inquirer.prompt([
+
+    transport = selectedTransport;
+  }
+
+  if (transport === 'stdio') {
+    const { mode } = await inquirer.prompt([
       {
-        type: 'input',
-        name: 'awsSecretAccessKey',
-        message: 'Enter AWS/Tigris Secret Access Key:',
+        type: 'list',
+        name: 'mode',
+        message: 'Run via:',
+        choices: supportedModes,
       },
     ]);
-    env.AWS_ACCESS_KEY_ID = awsAccessKeyId;
-    env.AWS_SECRET_ACCESS_KEY = awsSecretAccessKey;
-  }
 
-  const { awsEndpointUrl } = await inquirer.prompt([
-    {
-      type: 'input',
-      name: 'awsEndpointUrl',
-      message: 'Enter AWS/Tigris S3 Endpoint:',
-      default: 'https://fly.storage.tigris.dev',
-    },
-  ]);
-  env.AWS_ENDPOINT_URL_S3 = awsEndpointUrl;
-
-  if (command === 'docker') {
-    console.log(
-      'Pulling the Tigris MCP Server Docker image. This may take a few minutes...\n',
-    );
-    const downloadImage = spawnSync(
-      'docker',
-      ['image', 'pull', tigrisMcpServerImage],
-      { encoding: 'utf-8' },
-    );
+    const { useAwsProfiles } = await inquirer.prompt([
+      {
+        type: 'confirm',
+        name: 'useAwsProfiles',
+        message: 'Do you want to use AWS profiles?',
+        default: false,
+      },
+    ]);
+
+    const command = mode === supportedModes[0] ? 'npx' : 'docker';
+    const args =
+      mode === supportedModes[0]
+        ? ['-y', '@tigrisdata/tigris-mcp-server', 'run'] // npx command
+        : ['run']; // docker command
 
-    if (downloadImage.error) {
-      console.error(`Error pulling image: ${downloadImage.error.message}`);
-    } else if (downloadImage.stderr) {
-      console.error(`stderr: ${downloadImage.stderr}`);
+    let env: Record<string, string> = {};
+
+    if (useAwsProfiles) {
+      const { awsProfile } = await inquirer.prompt([
+        {
+          type: 'input',
+          name: 'awsProfile',
+          message: 'Enter AWS profile name:',
+          default: 'default',
+        },
+      ]);
+      env.USE_AWS_PROFILES = 'true';
+      env.AWS_PROFILE = awsProfile;
     } else {
-      console.log(`stdout: ${downloadImage.stdout}`);
+      const { awsAccessKeyId } = await inquirer.prompt([
+        {
+          type: 'input',
+          name: 'awsAccessKeyId',
+          message: 'Enter Tigris Access Key Id:',
+        },
+      ]);
+      const { awsSecretAccessKey } = await inquirer.prompt([
+        {
+          type: 'input',
+          name: 'awsSecretAccessKey',
+          message: 'Enter Tigris Secret Access Key:',
+        },
+      ]);
+      env.AWS_ACCESS_KEY_ID = awsAccessKeyId;
+      env.AWS_SECRET_ACCESS_KEY = awsSecretAccessKey;
```

Comment:
This syntax upserts into objects right?

Reply:
No it assigns the value, don't confused `env` here with Node runtime environment variables.

Reply:
ah yes it does okay cool